parachain/primitives/router/src/outbound/mod.rs

```diff
@@ -303,6 +365,43 @@ impl<'a, Call> XcmConverter<'a, Call> {
 		Ok(AgentExecuteCommand::TransferToken { token: asset, recipient: destination, amount })
 	}
 
+	fn transact_message(&mut self) -> Result<AgentExecuteCommand, XcmConverterError> {
+		use XcmConverterError::*;
+
+		log::trace!(target: "xcm::ethereum_blob_exporter", "before descend origin.");
+
+		let cmd = self.next()?;
+
+		let contract_address = if let DescendOrigin(location) = cmd {
+			if let X1(AccountKey20 { network: _token_network, key: contract_address }) = location {
+				contract_address
+			} else {
+				return Err(InvalidJunction)
+			}
+		} else {
+			return Err(DescendOriginExpected)
+		};
+
+		log::trace!(target: "xcm::ethereum_blob_exporter", "before transact.");
+
+		let data = if let Transact {
+			origin_kind: _origin_kind,
+			require_weight_at_most: _require_weight_at_most,
+			call,
+		} = self.next()?
+		{
+			call
+		} else {
+			return Err(TransactExpected)
+		};
+
+		Ok(AgentExecuteCommand::Transact {
+			target: contract_address.into(),
+			payload: data.encode(),
```

Comment:
@yrong The `data` to be sent to Ethereum is of type `DoubledEncoded<Call>`. I am struggling to get it into a `Vec<u8>` type. I have used `.encode()` here, but I do not think this is correct. It also prepends an extra byte to the data array, as seen by a test here: https://github.com/Snowfork/snowbridge/pull/924/files#diff-8de255c76ce8e2d34cf5bc5a3d0a46919ff8bf9f7af384d29a85dad048b95ec7R1701

Reply:
> prepends an extra byte to the data array

Not running the test yet but guess the extra byte is the length of the array which is how scale encode array. So we can just remove the extra byte here.